[Inductor] Refactor autotuning utils to compute max block sizes

blaine-rister opened the pull request:
These OSS changes are part of a larger MTIA diff. The OSS part is a simple refactor that makes it easier to query max block sizes by the prefix of the grid dimension, e.g. `"X"`, as opposed to having to use separate functions for `get_xmax()`, `get_ymax()`, etc. 

Differential Revision: D60195669


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

blaine-rister commented:
TODO: refactor the 1 failing test

shunting314 commented on torch/_inductor/runtime/coordinate_descent_tuner.py:
```diff
@@ -54,29 +54,20 @@ def __init__(
+        have_size_hint = (
+            self.size_hints is not None
+            and len(self.size_hints) > 0
+            and len(self.size_hints) > size_hint_idx
```
Looks good overall, but size_hint_idx value '-1' need to be specially handled (i.e. wrap around).

blaine-rister replied:
That one seems tricky. My thought here was, if `size_hint_idx == -1`, then we always have `len(self.size_hints) > size_hint_idx`. So in that case, this is just checking that we have any size hints at all.

shunting314 replied:
yea. Checking `len(self.size_hints) > 0` rather than `len(self.size_hints) > -1` looks more correct.

blaine-rister committed:
[Inductor-MTIA] bump up block sizes to prevent small loads/stores in Triton-MTIA (#131730)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/131730

Triton-MTIA likes to see loads/stores of at least 32 bytes. This diff attempts to modify Inductor's block sizes to achieve this. If we notice that Inductor is using a block size which would correspond to less than 32 bytes, we increase it.

In case raising the block size would violate Inductor's own assumptions, we log a warning.

Test Plan:
Added a unit test which intercepts Inductor's block size, changes it to a small number, then checks that our backend can detect this and raise it to 32 bytes.

`buck2 test @//mode/{opt,mtia,inplace} fbcode//triton_mtia/python/test:test_inductor_kernels -- bump`

Reviewed By: jfix71

Differential Revision: D60195669
